### PR TITLE
network: make Broadcast= accept "no"

### LINKED ISSUE #18492: systemd-networkd assigning broadcast address to WireGuard interfaces
`systemd-networkd` supports adding WireGuard interfaces and assigning IP addresses to them. That's very nice.
However, when assigning an IP address to the interface, a broadcast address is added too, as showed by the `ip addr` command. This happens if, for example, a /24 subnet mask is specified, which is useful because the kernel will add a routing table for the subnet. Anyway, it's my understanding that WireGuard doesn't support broadcasting.

When using `wg-quick` provided by wireguard-tools a broadcast address is in fact not added to the interface. Would it be possible to make `systemd-networkd` not assign a broadcast address to WireGuard interfaces too?

I'm submitting this issue as a feature request and not as a bug because nothing is actually broken and it's only an "aesthetic" request. I understand if this is not high priority but would it be nice if it was fixed nonetheless.

### PR DESCRIPTION
**Pull request by yuwata**
Closes #18492.

### CODE REVIEW

**Comment by keszybz** [comment]
Hmm, I'm not expert on this, but based on the comments, the broadcast address is not used:
> There is actually no subnet for WireGuard. WireGuard simply looks up the allowed IPs to determine which peer should the packet be forwarded to.

So why not make Broadcast=no the default when configuring a wg interface? Backwards compat is good, but if the behaviour is not useful for anything, we don't need to keep it.

**Comment by yuwata** [comment]
@keszybz Thank you for your comment. OK. Now `Broadcast=` accepts both `yes` and `no`. And if unspecified, broadcast will not be set automatically for wireguard interfaces. PTAL.

File: man/systemd.network.xml [hunk 1]
```diff
@@ -1044,12 +1044,13 @@ IPv6Token=prefixstable:2002:da8:1::</programlisting></para>
...
           <listitem>
-            <para>The broadcast address, which must be in the format
-            described in
+            <para>Takes an IPv4 address or boolean value. The address must be in the format described in
             <citerefentry project='man-pages'><refentrytitle>inet_pton</refentrytitle><manvolnum>3</manvolnum></citerefentry>.
-            This key only applies to IPv4 addresses. If it is not
-            given, it is derived from the <varname>Address=</varname>
-            key.</para>
+            If set to <literal>yes</literal>, then broadcast address will be derived from the
+            from the <varname>Address=</varname> setting is used. If set to <literal>no</literal>, then
```
Comment (keszybz) [inline man/systemd.network.xml]: "from the from the", also "is used" seems misplaced.

```
If set to true, then the IPv4 broadcast address will be derived from the
<varname>Address=</varname> setting. If set to false, then
the broadcast address will not be set.
Defaults to true, except for wireguard interfaces, where it default to false.
```

**Review by keszybz — COMMENTED** [review]
Looks nice, except for the doc wording.

**Comment by poettering** [comment]
makes sense, lgtm, too

**Comment by yuwata** [comment]
@keszybz and @poettering Thank you for you review. The man page is updated. Upgrading the green label.

**Pull request merged**